### PR TITLE
Add asynchronous hardware components documentation

### PR DESCRIPTION
**Pull request by saikishor**
Add missing docs from https://github.com/ros-controls/ros2_control/pull/1567

### CODE REVIEW

File: hardware_interface/doc/asynchronous_components.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The ``ros2_control`` framework allows to run hardware components asynchronously. This is useful when some of the hardware components need to run in a separate thread or executor. For example, a sensor takes longer to read data that affects the periodicity of the  ``controller_manager`` control loop. In this case, the sensor can be run in a separate thread or executor to avoid blocking the control loop.
+
+
+Examples
+***********
+
+The following examples show how to use the different hardware interface types synchronously and asynchronously with ``ros2_control`` URDF.
+They can be combined together within the different hardware component types (system, actuator, sensor) (:ref:`see detailed documentation <overview_hardware_components>`) as follows
+
+For a RRBot with Multimodal gripper and external sensor:
```
Comment (christophfroehlich) [inline hardware_interface/doc/asynchronous_components.rst:17]:
```suggestion
For a RRBot with multimodal gripper and external sensor:
```

**Review by christophfroehlich — CHANGES_REQUESTED** [review]
Should we list the parameters explicitely before they are introduced in the example? Please add a line to the release_notes as well.

File: doc/release_notes.rst [hunk 2]
```diff
@@ -158,6 +158,7 @@ hardware_interface
 * With (`#1421 <https://github.com/ros-controls/ros2_control/pull/1421>`_) a key-value storage is added to InterfaceInfo. This allows to define extra params with per Command-/StateInterface in the ``.ros2_control.xacro`` file.
 * With (`#1763 <https://github.com/ros-controls/ros2_control/pull/1763>`_) parsing for SDF published to ``robot_description`` topic is now also supported.
 * With (`#1567 <https://github.com/ros-controls/ros2_control/pull/1567>`_) all the Hardware components now have a fully functional asynchronous functionality, by simply adding ``is_async`` tag to the ros2_control tag in the URDF. This will allow the hardware components to run in a separate thread, and the controller manager will be able to run the controllers in parallel with the hardware components.
+* Add documentation for asynchronous hardware components (`#1961 <https://github.com/ros-controls/ros2_control/pull/1961>`_)
```
Comment (christophfroehlich) [inline doc/release_notes.rst:161]: Sorry, I haven't seen that the line before was added in the very end of 1567. There is no need to advertise the docs here.
```suggestion
```

**Review by christophfroehlich — APPROVED** [review]
Now it's perfect 👍

**Comment by saikishor** [comment]
> Now it's perfect 👍 

Thank you for fixing it

**Pull request merged**